alloc.c

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 OpenSWE1R Maintainers
+// Licensed under GPLv2 or any later version
+// Refer to the included LICENSE.txt file.
+
+// This is a temporary placeholder for a memory allocator.
+
+#include "alloc.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+static unsigned int block_count = 0;
+static uint32_t block_size = 0;
+static uint8_t* block_usage = NULL;
```

Comment:
Move this stuff into an Allocator object; that way, it can be re-used for the file-handles.

Block size can also be moved into caller (not sure yet, wether this makes sense).